lib/middleware/action-attach-metadata.js

```diff
@@ -0,0 +1,25 @@
+const { SlackWorkspace, SlackUser } = require('../models');
+
+/**
+ * Ensure that the current Slack user is associated with a GitHub user.
+ */
+module.exports = async function attachMetaData(req, res, next) {
+  const userId = req.body.user.id;
+  const teamId = req.body.team.id;
+
+  const slackWorkspace = await SlackWorkspace.findOne({ where: { slackId: teamId } });
+  const [slackUser] = await SlackUser.findOrCreate({
+    where: { slackId: userId, slackWorkspaceId: slackWorkspace.id },
+  });
+
+
+  // Store metadata in res.locals so it can be used later in the request
+  Object.assign(res.locals, {
+    slackUser,
+    slackWorkspace,
+    action: req.body.actions[0].name,
+    value: req.body.actions[0].value,
+  });
+
+  next();
+};
```

Comment:
Two ways in which this is different from the `authenticate` middleware:
1. (the big one): The `authenticate` middleware ensures that a GitHub account is connected (to an existing slackUser). This middleware creates a new slackUser if one doesn't exist, just to make the slackUser available on `res.locals`
2. The data format is slightly different between slash commands and actions. For example on slash commands a user id is `req.body.user_id`, whereas on actions it's `req.body.user.id`

> Also I'm not sure what action-attach-metadata means in the context of authentication/authorization.

I'm not sure if what the middleware does is really authentication/authorization. It's kind of just a helper that makes the model available so that each "action route" doesn't have to worry about it.

Reply:
> Maybe one to look up the current workspace/user, and the other just verifies that the current user has a GitHub user associated?

I agree this would be good to do 👍